save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/execution/ExecutionDto.kt

```diff
@@ -42,6 +42,33 @@ data class ExecutionDto(
     val testSuiteSourceName: String?,
     val score: Double?,
 ) {
+    /**
+     * @param numerator
+     * @param denominator
+     * @return rate based on [numerator] and [denominator]
+     */
+    @Suppress("MagicNumber")
+    fun calculateRate(numerator: Long, denominator: Long) = denominator.takeIf { it > 0 }
+        ?.run { numerator.toDouble() / denominator }
+        ?.let { it * 100 }
+        ?.toInt()
+
+    /**
+     * @return precision rate
+     */
+    fun getPrecisionRate() = calculateRate(matchedChecks, matchedChecks + unexpectedChecks) ?: 0
+
+    /**
+     * @return recall rate
+     */
+    fun getRecallRate() = calculateRate(matchedChecks, matchedChecks + unmatchedChecks) ?: 0
+
+    /**
+     * @return F-measure
+     */
+    // FixMe: do we need some other algorithm? for now it's F-measure
+    fun calculateScore() = (2 * getPrecisionRate() * getRecallRate()) / (getPrecisionRate() + getRecallRate()).toDouble()
```

Comment:
I think that to support better extensibility in the future, we should from the beginning isolate this logic a bit more, maybe even move completely to JVM sources and only request already computed values from the frontend. Then we may introduce a enum of score types, for example. Also `calculateScore` should accept at least `TestingType` as an argument, because I would imagine that score measures would differ for contests and for ordinary executions.
cc @akuleshov7

Reply:
Moved into separate file, added enum for score types